src/cli/CliRandomizer.java

```diff
@@ -0,0 +1,194 @@
+        }
+        if (settings.isUpdatedFromOldVersion()) {
+            System.out.println(bundle.getString("GUI.settingsFileOlder") + "%n");
+        }
+    }
+
+    public static int invoke(String[] args) {
+        String settingsFilePath = null;
+        String sourceRomFilePath = null;
+        String outputRomFilePath = null;
+        boolean saveAsDirectory = false;
+
+        List<String> allowedFlags = Arrays.asList("-i", "-o", "-s", "-d");
+        for (int i = 0; i < args.length; i++) {
+            if (allowedFlags.contains(args[i])) {
+                int flagValueIndex = i + 1;
+
+                switch(args[i]) {
+                    case "-i":
+                        sourceRomFilePath = args[i + 1];
+                        break;
+                    case "-o":
+                        outputRomFilePath = args[i + 1];
+                        break;
+                    case "-s":
+                        settingsFilePath = args[i + 1];
+                        break;
+                    case "-d":
+                        saveAsDirectory = true;
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        if (settingsFilePath == null || sourceRomFilePath == null || outputRomFilePath == null) {
+            System.err.println("Missing required argument");
+            return 1;
+
+        }
+
+        // now we know we have the right number of args...
+        if (!new File(settingsFilePath).exists()) {
+            System.err.println("Error: could not read settings file");
+            System.out.println(CliRandomizer.usage());
+            return 1;
+        }
+
+        // check that everything is readable/writable as appropriate
+        if (!new File(sourceRomFilePath).exists()) {
+            System.err.println("Error: could not read source ROM file");
+            System.out.println(CliRandomizer.usage());
+            return 1;
+        }
+
+        // java will return false for a non-existent file, have to check the parent directory
+        if (!new File(outputRomFilePath).getAbsoluteFile().getParentFile().canWrite()) {
+            System.err.println("Error: destination ROM path not writable");
+            System.out.println(CliRandomizer.usage());
+            return 1;
+        }
+
+        boolean processResult = CliRandomizer.performDirectRandomization(
+                settingsFilePath,
+                sourceRomFilePath,
+                outputRomFilePath,
+                saveAsDirectory
+        );
+        if (!processResult) {
+            System.err.println("Error: randomization failed");
+            System.out.println(CliRandomizer.usage());
+            return 1;
+        }
+        return 0;
+    }
+
+    private static String usage() {
+        return "Usage: java -jar PokeRandoZX.jar cli -s <path to settings file> -i <path to source ROM> -o <path for new ROM> [-d save as directory]";
```

Comment:
I'd rephrase this just to make it clear that the `-d` flag should be on its own, without anything extra after it. Also, to make it a bit simpler, `usage()` can just print directly instead of returning the string.
```suggestion
        System.err.println("Usage: java -jar PokeRandoZX.jar cli -s <path to settings file> -i <path to source ROM> -o <path for new ROM> [-d]");
        System.err.println("-d: Save 3DS game as directory (LayeredFS)");
```